resources/js/components/FormField.vue

```diff
@@ -89,16 +90,16 @@ export default {
          */
         setInitialValue() {
             this.items = Array.isArray(this.field.value) ? this.field.value : [];
-            this.items = this.items.map(item=>{
-                return { 'id': this.getNextId(), 'fields':item }
+            this.items = this.items.map((item, index) => {
+                return { 'id': this.getNextId(), 'modelId': this.field.models[index], 'fields':item }
```

Comment:
Let's put each key on its own line
```suggestion
                return { 
                    'id': this.getNextId(),
                    'modelId': this.field.models[index],
                    'fields':item
                };
```